File: php/postorderTraversal.php
<?php

class TreeNode
{
    public $val = null;
    public $left = null;
    public $right = null;

    function __construct($val = 0, $left = null, $right = null)
    {
        $this->val = $val;
        $this->left = $left;
        $this->right = $right;
    }
}

/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param TreeNode $root
     * @return Integer[]
     */
    function postorderTraversal($root) {
        $orders = [];
        if (is_null($root)) {
            return $orders;
        }

        if (!is_null($root->left)) {
            $orders = array_merge($orders, $this->postorderTraversal($root->left));
        }
        if (!is_null($root->right)) {
            $orders = array_merge($orders, $this->postorderTraversal($root->right));
        }
        $orders[] = $root->val;
        return $orders;
    }
}

var_dump(
    (new Solution())->postorderTraversal(
        (new TreeNode(
            1,
            null,
            (new TreeNode(
                2,
                (new TreeNode(3))
            ))
        ))
    )
);

File: go/postorderTraversal.go
package main

type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func postorderTraversal(root *TreeNode) []int {
    var result []int
    var stack [] *TreeNode

    for ;; {
        if root == nil {
            stackSize := len(stack)
            if stackSize > 0 {
                root = stack[stackSize - 1]
                stack = stack[:stackSize - 1]
            }
        }

        if root == nil {
            break
        }

        if root.Left != nil {
            leftNode := root.Left
            root.Left = nil
            stack = append(stack, root)
            root = leftNode
        } else if root.Right != nil {
            rightNode := root.Right
            root.Right = nil
            stack = append(stack, root)
            root = rightNode
        } else {
            result = append(result, root.Val)
            root = nil
        }
    }

    return result
}
